include/evmc/evmc.hpp

```diff
@@ -827,31 +815,25 @@ namespace std
 template <>
 struct hash<evmc::address>
 {
-    /// Hash operator using FNV1a-based folding.
+    /// Hash operator using (3a + b) folding of the address "words".
```

Comment:
What is `3a + b`? Some homebrew hashing?

Reply:
Kindof. We have this progression of options:
1. Fold all words with XOR.
2. Fold all words with ADD. A bit better than XOR because discards less information. But also symmetric.
3. "Classic" multiply by prime/odd number and add: `fold(a,b) { return 3*a + b }`.

The 3 is used because has the same performance as 1 and 2. The multiply is done by `lea` instruction and the throughput is the same because of the executing multiple instructions in the same time. I.e. latency of "multiply" is hidden.

Reply:
How would 1) and 2) work? The same bytes in a different order would result in the same hash. Or do you mean not only xor/add, but some shifting/etc.

Reply:
Just `word0 ^ word1 ^ word2 ^ word3`. Similarly add.

Reply:
If I remember correctly, we discussed that this is only used as a quick lookup, but the actual data is then compared at a match, so clashes do not matter.